### PR TITLE
Added a tutorial about starting with launchpad

### LINKED ISSUE #11: As a new user, I want a tutorial on how to build my first plugin
**Context**

While the documentation provides good explanations of each component and concepts of the library, as a new user, I don't know where to start and what knowledge is important or not. The only obvious documentation file I could find was general/installation.md. But then, I don't know what to do. Reading the files, it's not clear if I should use a Module, a container, or a subscriber, etc.

**What needs to be done**
Add a "Getting started" section in the documentation, with a step by step demonstration of how to create a basic plugin that:
- performs a dummy action on a given hook.
- has a test.
Then, add steps to covers AS, DB, Logger, FS, front-end, uninstaller and hook-extractor.

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

Added a documentation on how to start with Launchpad.

Fixes #11

## Type of change

- [ ] This change requires a documentation update

## Is the solution different from the one proposed during the grooming?

Please describe in this section if there is any change to the solution, and why.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

Please delete the options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by CrochetFeve0251** [comment]
@wp-launchpad/wp-media if anyone can review it?

File: wiki/general/tutorials/getting-started-.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+Once the command has been executed, it should ask you for your plugin name, and it will generate the necessary changes from it.
+
+You can assert the command executed successfully by checking if the name from your plugin file matches what was the name from your plugin.
+
+### Install a module in your project using composer
+
+Now that the project is set up properly, it is now time to install some modules to gain time and effort when working on your project.
+
+A good option would be to use the WordPress options facade to inside the module `wp-launchpad/framework-options`.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: A part is missing in this sentence? It would be good to explain quickly what this module does.

File: wiki/general/tutorials/getting-started-.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+For that we will need to install the corresponding take off package `wp-launchpad/framework-options-take-off` that will set it up automatically for us.
+
+For that we will have to run the following command: `composer install wp-launchpad/framework-options-take-off`
+
+Once this is done, then you can assert everything went as expected by looking that the provider has been added to `configs/providers.php` and that we have the matching package installed in the `composer.json` file.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: Small example? What provider are we looking for?

File: wiki/general/tutorials/getting-started-.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+It is possible to find a list of every module available on [the module list page](../../modules/listing.md).
+
+### Create a new subscriber
+
+Once the project is now set up properly.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: Is this sentence complete?

File: wiki/general/tutorials/getting-started-.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+It is now a great time to create our first subscriber.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: Maybe a link to what a subscriber is?

File: wiki/general/tutorials/getting-started-.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+
+#### Create a new subscriber
+
+To create a subscriber, we can create any class with the name of your choice.
+
+What will make it become a subscriber is actually adding a docblock with the annotation `@hook`.
+
+This annotation signals that this method should be used as an event handler for some event.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md:70]: Is there a more detailed documentation about this syntax? If yes, then link it. Otherwise, it will be needed: for someone coming from "normal" WP, this is a big change and it's not clear how this replaces add_action for instance.

File: wiki/general/tutorials/getting-started-.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+}
+```
+
+#### Use inflectors
+Inflectors are a key tool when working with Launchpad plugins and services.
+These allow you to access some features without having to bind them directly to your code.
+
+In our case we're going to use one to load the option facade from the module we installed earlier on.
+
+For that on our subscriber we will have to first implement the interface `` and use the trait ``.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: Interface and trait names missing?

**Review by MathieuLamiot — COMMENTED** [review]
Same feedback as on the other PR:
- Maybe review by a native speaker? Or ask AI to rephrase the document.
- Consider linking a "real" example.

**Comment by CrochetFeve0251** [comment]
> * Maybe review by a native speaker? Or ask AI to rephrase the document.

@MathieuLamiot I used IA in that PR to speed up the writing process. :')

**Comment by MathieuLamiot** [comment]
> @MathieuLamiot I used IA in that PR to speed up the writing process. :')

From my experience, it can give good result to provide the whole text to an AI once it is written (even if originally written with an AI) to put a final touch to it. For instance, I copy/pasted this Getting Started to ChatGPT telling it to rephrase it to make it sound like a professional documentation, and the results are good I think.

**Comment by CrochetFeve0251** [comment]
> > @MathieuLamiot I used IA in that PR to speed up the writing process. :')
> 
> From my experience, it can give good result to provide the whole text to an AI once it is written (even if originally written with an AI) to put a final touch to it. For instance, I copy/pasted this Getting Started to ChatGPT telling it to rephrase it to make it sound like a professional documentation, and the results are good I think.

@MathieuLamiot I reworked the tutorial to add fix mentioned problems.

File: wiki/general/tutorials/getting-started-.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+There are [multiple methods available](../../container/providers.md) to load them depending on the context but in this tutorial we will add them to `get_common_subscribers` method which load the subscribers in any context.
+
+Inside the `inc` folder from your plugin a default service provider has been created in the file `inc/ServiceProvider.php`.
+
+To register the subscriber we need to add the following code:
+
+```php
+public function get_common_subscribers(): array {
+    return [
+        MySubscriber::class;
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: There is a typo here. It should be , instead, no?

File: wiki/general/tutorials/getting-started-.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+### Install a module in your project using composer
+
+Now that the project is set up properly, it is now time to install some modules to gain time and effort when working on your project.
+
+A good option to start with would be to use the WordPress options facade to inside the module `wp-launchpad/framework-options`.
+
+For that we will need to install the corresponding take off package `wp-launchpad/framework-options-take-off` that will set it up automatically for us.
+
+For that we will have to run the following command: `composer install wp-launchpad/framework-options-take-off`
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md]: Shouldn't it be composer require?

File: wiki/general/tutorials/getting-started-.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+Once this is completed, you should be able to see a new folder with the code for your project inside it. However, it is not yet ready to be used as it is still Launchpad default code and needs to be initialized.
+
+#### Initialize project with launchpad command
+
+As multiple projects can exist on the same WordPress website at once, it is important to containerize all of them, and for that purpose we will use a specific namespace for our code.
+
+To do so with Launchpad, we need to run the following command in the terminal:
+`bin/generator initialize`
+
+Once the command has been executed, it should ask you for your plugin name, and it will generate the necessary changes from it.
```
Comment (MathieuLamiot) [inline wiki/general/tutorials/getting-started-.md:19]: Giving "debug-hooks" as an input raises a PSR-4 error. However, putting "Debug Hooks" is properly handled and generates file named "debug-hooks" and namespace "DebugHook". It's worth mentioning that the display name must be given, and that the rest is taken care of.
In my case, I never imagined I could put a name with a whitespace.

**Review by MathieuLamiot — APPROVED** [review]
Ok on my side.
